client/components/ProductDetail/UpdateCart.jsx

```diff
@@ -1,64 +1,86 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import styled from 'styled-components';
-import AddToCart from './AddToCart.jsx';
+import ls from 'local-storage';
+import { Button } from '../Shared/Form';
 
-const UpdateCart = (props) => {
-  var availability = [{
-    size: 'Select Size',
-    quantity: null
-  }];
+const UpdateCart = ({ style, product, addToCart }) => {
+  //Define Variables for Functional Component
+  const [selectedSize, setSelectedSize] = useState(0);
+  const [selectedQuantity, setSelectedQuantity] = useState(0);
+  const [warning, setWarning] = useState(false);
+  let sizes;
+  let availableQuantities;
+  let quantities;
 
-  for (var key in props.currentStyle.skus) {
-    if (!props.currentStyle.skus[key].quantity) {
-      continue;
+  //If the user switches the style, then reset the state
+  useEffect(() => {
+    setSelectedSize(0);
+    setSelectedQuantity(0);
+  }, [style]);
+
+  //Handle a click on the cart
+  let handleAddCartClick = (validation) => {
+    console.log(selectedSize);
+    if (selectedSize === 0 && validation === true) {
+      setWarning(true);
+    } else {
+      addToCart({
+        product_id: product.id,
+        style_id: style ? style.style_id : null,
+        sku_id: style ? Object.keys(style.skus)[selectedSize] : null,
+        size: style && selectedSize > 0 ? sizes[selectedSize] : null,
+        quantity: style && selectedSize > 0 ? availableQuantities[selectedQuantity] : null,
```

Comment:
ya id say by this point theres a lot of boolean resolvers here that they deserve aliasing. if something goes wrong then you're just reading through a bunch of compounded boolean logic. its easier to just alias these out.

```node.js
const hasQuantity = style 
  && selectedSize > 0 
  && selectedQuantity
  && availableQuantities[selectedQuantity]
const quantity = hasQuantity ? availableQuantities[selectedQuantity] : null
```